test/test_custom_ops.py

```diff
@@ -623,6 +623,11 @@ def g(
             infer_schema(g), """(Tensor x, Tensor[] y, Tensor[] z, Tensor?[] w) -> ()"""
         )
 
+        self.assertExpectedInline(
+            infer_schema(g, mutated_args={"x", "w", "z"}),
+            """(Tensor(a0!) x, Tensor[] y, Tensor(a2!)[] z, Tensor(a3!)?[] w) -> ()""",
```

Comment:
> Is someone allowed to return an output that aliases an input today?

Yes, using the existing torch.library APIs, but we don't recommend it because (1) there's no way to easily provide a functionalization kernel for it (2) the user almost always can rewrite their custom op to not do this.